### PR TITLE
[EXPERIMENT] feat(@angular-devkit/build-angular): optimize names of internal AOT properties

### PR DESCRIPTION
**Pull request by clydin**

### CODE REVIEW

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts [hunk 1]
```diff
@@ -411,21 +411,31 @@ export function getCommonConfig(wco: WebpackConfigOptions): Configuration {
...
+      mangle: allowMangle &&
+        buildOptions.platform !== 'server' &&
+        !differentialLoadingMode && {
+          properties: {
+            regex: /^(?:ɵfac|ɵcmp|ɵinj|ɵmod)$/,
+          },
+        },
+      nameCache: {
+        vars: { props: {} },
+        props: { props: { $ɵmod: 'm', $ɵinj: 'i', $ɵfac: 'f', $ɵcmp: 'c' } },
```
Comment (petebacondarwin) [inline packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: What happens if a component (say) already has a property called `c`?

Comment (alan-agius4) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: These changes would also brea analytics build stats

Comment (clydin) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: The existing static property gets overwritten by the Angular property.  This is the same as current behavior (although less likely with the more exotic name) and is really a consequence of using string property names.  To fix this we could consider using symbol names instead.  They work well as a metadata store.

Comment (petebacondarwin) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: Yeah but `ɵcmp` is much less likely to be used on a component than `c`...

Comment (clydin) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: I did mention that :)

Alternatively, we could try running terser serially and let it pick a unique name; although at the cost of build time.

Comment (alfaproject) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: I can see the appeal but should we really try to reinvent compression? Surely, everyone delivers static assets compressed by gzip or brotli or...

Comment (alfaproject) [reply packages/angular_devkit/build_angular/src/angular-cli-files/models/webpack-configs/common.ts:437]: I see, thank you for expanding. (:

**Pull request closed without merging**